src/NodeParser/StringTemplateLiteralNodeParser.ts

```diff
@@ -0,0 +1,73 @@
+import ts from "typescript";
+import { UnknownTypeError } from "../Error/UnknownTypeError";
+import { Context, NodeParser } from "../NodeParser";
+import { SubNodeParser } from "../SubNodeParser";
+import { AliasType } from "../Type/AliasType";
+import { BaseType } from "../Type/BaseType";
+import { LiteralType } from "../Type/LiteralType";
+import { UnionType } from "../Type/UnionType";
+
+export class StringTemplateLiteralNodeParser implements SubNodeParser {
+    public constructor(protected childNodeParser: NodeParser) {}
+
+    public supportsNode(node: ts.NoSubstitutionTemplateLiteral | ts.TemplateLiteralTypeNode): boolean {
+        return (
+            node.kind === ts.SyntaxKind.NoSubstitutionTemplateLiteral || node.kind === ts.SyntaxKind.TemplateLiteralType
+        );
+    }
+    public createType(node: ts.NoSubstitutionTemplateLiteral | ts.TemplateLiteralTypeNode, context: Context): BaseType {
+        if (node.kind === ts.SyntaxKind.NoSubstitutionTemplateLiteral) {
+            return new LiteralType(node.text);
+        }
+        const prefix = node.head.text;
+        const matrix: string[][] = [[prefix]];
+
+        for (const span of node.templateSpans) {
+            const suffix = span.literal.text;
+
+            const type = this.childNodeParser.createType(span.type, context);
+            const literals = [...extractLiterals(type)];
+
+            matrix.push(literals.map((value) => value + suffix));
```

Comment:
Would it be easier to replace this for loop with a map over `node.templateSpans` that you assign to `matrix`?

Reply:
I can make it a map.

Reply:
done.